### PR TITLE
Fix NIC hotplug with `vhost_net=True`

### LINKED ISSUE #1651: NIC hotplug with vhost_net=True broken
While trying to investigate #1649 I discovered an other issue regarding NIC hotplug. When the KVM HV parameter `vhost_net` is set to=`True`, NIC-Hotplug fails with:
```
# Ganeti=https://github.com/ganeti/ganeti:master OS=debian-bullseye 
root@master:~# gnt-instance modify --hotplug --net add:network=vm-net test.vm
Sun Feb  6 11:50:42 2022  - INFO: Trying to hotplug device...
Sun Feb  6 11:50:42 2022  - WARNING: Could not hotplug device: Error while executing backend function: kvm: error executing the netdev_add command: Invalid parameter type for 'vhost', expected: boolean (GenericError):
```
According [this Issue](https://bugzilla.redhat.com/show_bug.cgi?id=1832710#c2), Qemu switched `netdev_add` to a type-safe representation. Ganeti currently sends `"vhost": "on"` as string: https://github.com/ganeti/ganeti/blob/7382df3b1494b1247cd8767d1e740f13f46b729a/lib/hypervisor/hv_kvm/monitor.py#L504-L508
Changing to `"vhost": True` solved the problem.

There might be more issues regarding types in QMP.

@rbott is `vhost_net` enabled in the QA-Suite?

At least no one should use virtio-networking without vhost.

### PR DESCRIPTION
**Pull request by saschalucas**
Since Qemu commit https://github.com/qemu/qemu/commit/db2a380c84574d8c76d7193b8af8535234fe5156 the QMP command `netdev_add` is enforced to have correct types. Effected versions are Qemu >=4.3 (Debian Bullseye and beyond). Fixing this by replacing the string with a Boolean.

This fix seems compatible down to Ubuntu Bionic (qemu-2.11) and should therefore be safe for backporting to 3.0.

Fixes #1651

### CODE REVIEW

**Comment by rbott** [comment]
Thanks for uncovering this @saschalucas. LGTM!

**Review by rbott — APPROVED** [review]

**Comment by rbott** [comment]
Just a quick sidenote: `vhost_net` seems to be completely undocumented in Ganeti. Maybe this PR could also add some details to the `gnt-instance` manpage?

We should even consider setting it to `true` as a sane default, but that should probably be at least a separate commit so we can choose to not bring that specific change into 3.0.2.

**Comment by saschalucas** [comment]
> Just a quick sidenote: `vhost_net` seems to be completely undocumented in Ganeti. Maybe this PR could also add some details to the `gnt-instance` manpage?

Thanks for looking at that detail. I added a manual page entry.

> We should even consider setting it to `true` as a sane default, but that should probably be at least a separate commit so we can choose to not bring that specific change into 3.0.2.

At least for me, defaulting `vhost_net` to `True` makes sense. Qemu seems to ignore the resulting CLI option `vhost=on` for NIC types other than paravirtual (i.e. e1000). So I added a separate commit here. It may be noticed, that the new default only applies to new clusters (`gnt-cluster init`), because `config.data` values are filled with defaults at cluster creation time.

To make `py-tests` happy, it was necessary to add a new `test/data/kvm_5.2.0_help.txt`. I also added a symlink `test/data/kvm_current_help.txt` and referenced the symlink inside the unit test. This way, we can later switch all tests at once to new test data.

@rbott PTAL (re-approve).

File: man/gnt-instance.rst [hunk 1]
```diff
@@ -912,6 +912,17 @@ kvm\_path
...
+vhost\_net
+    Valid for the KVM hypervisor.
+
+    This boolean option determines whether the tap devices used by the KVM
+    paravirtual nics (virtio-net) will use accelerated data plane, passing
+    network packets directly between host and guest kernel, without going
+    through userspace emulation layer (qemu).
+
+    Histrionically it is set to ``false`` by default, but everyone is
+    encouraged to enable it.
```
[the remark this answers is not in the document]
Comment (saschalucas) [reply man/gnt-instance.rst]: Thanks @mbakke for your review. I've fixed the typo and mentioned the new default for Ganeti-3.1.

**Comment by rbott** [comment]
Looks good, please proceed @saschalucas :-)

**Comment by rbott** [comment]
...ok one more question :-) 

Should we change this (a few lines below) as well to `True`?
https://github.com/ganeti/ganeti/blob/6b6e28276447880f85a37c046682e0154f056b43/lib/hypervisor/hv_kvm/monitor.py#L522-L526

**Comment by rbott** [comment]
Thanks for researching the "mq" part - I should have done that myself :-(

**Pull request merged**